### ISSUE TITLE
Modal IGT flags 3 issues on native dialog

### ISSUE THREAD

**Issue by joppekroon** [dequelabs/axe-core#4310]
### Product

axe Extension

### Product Version

v4.73.1

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
No issues reported

#### Actual
The Modal Dialog IGT on a native <dialog> reports 3 issues:

1. Dialog / Alert Dialog: Dialog is missing appropriate role and/or attributes
2. Able to browse outside modal with screen reader
3. Keyboard focus is not maintained in modal

#### How to Reproduce
Sample page can be found at https://ch3pyf.csb.app/, or try the code below.

```html
<body>
  <h1>Lets try a native dialog</h1>
  <button id="invoker">Show dialog</button>

  <dialog id="dialog" aria-labelledby="dialog-title">
    <h1 id="dialog-title">Hello Dialog</h1>
    <form method="dialog">
      <button>Close dialog</button>
    </form>
  </dialog>

  <script>
    const invoker = document.querySelector("#invoker");
    const dialog = document.querySelector("#dialog");

    invoker.addEventListener("click", () => {
      dialog.showModal();
    });
  </script>
</body>
```

I believe, in this setup, the dialog is not missing appropriate role or attributes, these are supplied natively.

And the other two issues are also not valid in my opinion, as the user is still correctly blocked from interacting with the _webpage_. With this native implementation the user is able to "escape" to the browser controls, this is a feature of the native dialog and an improvement on the old solutions where the user was completely held hostage by the dialog and cannot interact with their computer until they deal with the dialog.

#### Additional context
When used in an iframe context, the native dialog does seem to only block interaction with that iframe and not the rest of the webpage. This means, that when a native modal dialog appears in an embedded iframe you can "escape" to the surrounding page by keyboard, however, you can also interact with the surrounding page by mouse. So this probably still shouldn't be considered an issue.

**Comment by straker** [comment]
Thanks for the issue. We've let the appropriate team know.

**Comment by joppekroon** [comment]
Is there an update on this issue?

**Comment by Amandeque** [comment]
Hello @joppekroon, I have linked this issue to our backlog ticket for the Axe Extension. We have prioritized it for our next development cycle, and it will be resolved in one of our upcoming releases. Thank you for your patience.

**Comment by Amandeque** [comment]
@joppekroon The issue has been resolved now. We’d love to hear any feedback you might have!

**Comment by joppekroon** [comment]
@Amandeque Very glad this has finally been resolved now, thank you!
